src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransPatterns.java

```diff
@@ -488,20 +490,35 @@ private void handleSwitch(JCTree tree,
                                                                     List.of(new WildcardType(syms.objectType, BoundKind.UNBOUND,
                                                                                              syms.boundClass)),
                                                                     syms.classType.tsym)));
+
+            // The boolean flag `hasUnconditionalPattern` is true when either there
```

Comment:
A it surprised that something has changed here - can you expand?

Reply:
Indeed that was not needed anymore.